cvxpy/reductions/solvers/conic_solvers/scipy_conif.py

```
@@ -97,20 +120,44 @@ def apply(self, problem):
     def solve_via_data(self, data, warm_start: bool, verbose: bool, solver_opts, solver_cache=None):
         from scipy import optimize as opt
 
+        # Check if the problem is a MIP and for SciPy version support.
+        problem_is_a_mip = data[s.BOOL_IDX] or data[s.INT_IDX]
+        scipy_supports_mip = not (Version(scipy.__version__) < Version('1.9.0'))
+
+        if problem_is_a_mip and not scipy_supports_mip:
+            raise ValueError("Solving MIPs requires SciPy version >= 1.9")
+
```

This should be handled by ``accepts()``.

---

How should the version checks be handled if you want me to update and use the `ConicSolver` implementation of `accepts`? It seems like a custom `accepts` implementation will be required if you want to retain backward compatibility with older versions of SciPy (which cannot solve MIPs). Same concern with removing the warning in your comment below this one: do you prefer backward compatibility, or do you want to assume compatible SciPy versions? (FYI: SciPy 1.9.0 was released yesterday.)

---

``ConicSolver.accepts`` can use the following method in the ``ParamConeProg`` class to check if the problem is mixed-integer:
https://github.com/tasseff/cvxpy/blob/c4ba1ab85ae1dbb6967df2ec1e54f843479bd330/cvxpy/reductions/dcp2cone/cone_matrix_stuffing.py#L155-L158
The ``SCIPY(ConicSolver)`` class will be responsible for correctly setting ``self.MIP_CAPABLE`` (which it does in the current PR's code). Assuming that value is correctly set then a generic solver-independent implementation of ``ConicSolver.accepts`` should work fine. Does that sound about right to you?

---

Regarding the error message, my objection is that the message is raised even for SciPy versions 1.9.0 if someone doesn't explicitly specify the solve method. I'm of the opinion that if SciPy >= 1.9.0 is installed then we should silently default to HiGHS.

CVXPY has a logging feature that operates on top of the logging features of numerical solvers. If this message is displayed then I suggest it be part of that log. That way someone would see the message if they solve with ``prob.solve(verbose=True)``.